src/ansys/dpf/core/data_sources.py

```diff
@@ -77,9 +78,12 @@ def __init__(self, result_path=None, data_sources=None, server=None):
                 self._internal_obj = core_api.data_processing_duplicate_object_reference(
                     data_sources
                 )
-            else:
+            elif hasattr(data_sources, "DESCRIPTOR") or isinstance(data_sources, int):
```

Comment:
You could also use if (hasattr(data_sources, "__module__") and data_sources.__module__ == 'data_sources_pb2')

Reply:
Note: int is for the case CLayer server, ctypes.c_void_p is returned (which is an int)